cmd/lotus-storage-miner/init.go

```diff
@@ -625,9 +623,32 @@ func createStorageMiner(ctx context.Context, api lapi.FullNode, peerid peer.ID,
 	} else if cctx.Bool("create-worker-key") { // TODO: Do we need to force this if owner is Secpk?
 		worker, err = api.WalletNew(ctx, types.KTBLS)
 	}
-	// TODO: Transfer some initial funds to worker
 	if err != nil {
-		return address.Undef, err
+		return address.Address{}, err
+	}
+
+	// make sure the worker account exists on chain
+	_, err = api.StateLookupID(ctx, worker, types.EmptyTSK)
+	if err != nil {
+		signed, err := api.MpoolPushMessage(ctx, &types.Message{
+			From:  owner,
+			To:    worker,
```

Comment:
If the worker is the owner this is a self transference then, will that be processed (the actor created) normally?

Reply:
When that happens, and we manage to get here, we'll likely fail as that would mean that the owner key doesn't exist on chain either